extensions/mssql/src/objectManagement/ui/objectManagementDialogBase.ts

```diff
@@ -0,0 +1,260 @@
+}
+
+function getDialogName(type: NodeType, isNewObject: boolean): string {
+	return isNewObject ? `New${type}` : `${type}Properties`
+}
+
+export abstract class ObjectManagementDialogBase<ObjectInfoType extends ObjectManagement.SqlObject, ViewInfoType extends ObjectManagement.ObjectViewInfo<ObjectInfoType>> {
+	protected readonly disposables: vscode.Disposable[] = [];
+	protected readonly dialogObject: azdata.window.Dialog;
+	protected readonly contextId: string;
+	private _viewInfo: ViewInfoType;
+	private _originalObjectInfo: ObjectInfoType;
+
+	constructor(private readonly objectType: NodeType,
+		docUrl: string,
+		protected readonly objectManagementService: IObjectManagementService,
+		protected readonly connectionUri: string,
+		protected isNewObject: boolean,
+		protected readonly objectName: string | undefined = undefined,
+		protected readonly objectExplorerContext?: azdata.ObjectExplorerContext,
+		dialogWidth: azdata.window.DialogWidth = 'narrow') {
+		const objectTypeDisplayName = getNodeTypeDisplayName(objectType, true);
+		const dialogTitle = isNewObject ? NewObjectDialogTitle(objectTypeDisplayName) : ObjectPropertiesDialogTitle(objectTypeDisplayName, objectName);
+		this.dialogObject = azdata.window.createModelViewDialog(dialogTitle, getDialogName(objectType, isNewObject), dialogWidth);
+		this.dialogObject.okButton.label = OkText;
+		this.disposables.push(this.dialogObject.onClosed(async () => { await this.dispose(); }));
+		const helpButton = azdata.window.createButton(HelpText, 'left');
+		this.disposables.push(helpButton.onClick(async () => {
+			await vscode.commands.executeCommand('vscode.open', vscode.Uri.parse(docUrl));
+		}));
+		this.dialogObject.customButtons = [helpButton];
+		this.contextId = generateUuid();
+		this.dialogObject.registerCloseValidator(async (): Promise<boolean> => {
+			const confirmed = await this.onConfirmation();
+			if (!confirmed) {
+				return false;
+			}
+			return await this.runValidation();
+		});
+	}
+
+	protected abstract initializeData(): Promise<ViewInfoType>;
+	protected abstract initializeUI(): Promise<void>;
+	protected abstract onComplete(): Promise<void>;
+	protected abstract onDispose(): Promise<void>;
+	protected abstract validateInput(): Promise<string[]>;
+
+	protected onObjectValueChange(): void {
+		this.dialogObject.okButton.enabled = JSON.stringify(this.objectInfo) !== JSON.stringify(this._originalObjectInfo);
+	}
+
+	protected async onConfirmation(): Promise<boolean> {
+		return true;
+	}
+
+	protected get viewInfo(): ViewInfoType {
+		return this._viewInfo;
+	}
+
+	protected get objectInfo(): ObjectInfoType {
+		return this._viewInfo?.objectInfo;
+	}
+
+	protected get originalObjectInfo(): ObjectInfoType {
+		return this._originalObjectInfo;
+	}
+
+	public async open(): Promise<void> {
+		await vscode.window.withProgress({
+			location: vscode.ProgressLocation.Notification,
+			title: LoadingDialogText
```

Comment:
Not a huge fan of the toast being used for this - is there a reason you're not showing the dialog immediately and then just having a loading spinner while it loads? We do that pretty much everywhere else.

Reply:
I am planning to do it in a follow up PR.